src/Command/Push/PushArtifactCommand.php

```diff
@@ -0,0 +1,284 @@
+      $this->sanitize($output_callback, $artifact_dir);
+      $this->checklist->completePreviousItem();
+    }
+
+    $this->checklist->addItem("Committing changes (commit hash: $commit_hash)");
+    $this->commit($output_callback, $artifact_dir, $commit_hash);
+    $this->checklist->completePreviousItem();
+
+    if (!$input->getOption('dry-run')) {
+      $this->checklist->addItem("Pushing changes to <options=bold>{$environment->vcs->path}</> branch in the <options=bold>{$environment->name}</> environment");
+      $this->push($output_callback, $artifact_dir);
+      $this->checklist->completePreviousItem();
+    }
+    else {
+      $this->logger->warning("The <options=bold>--dry-run</> option prevented changes from being pushed to Acquia Cloud. The artifact has been built at <options=bold>$artifact_dir</>");
+    }
+
+    return 0;
+  }
+
+  /**
+   * Prepares a directory to build the artifact.
+   *
+   * @param \Closure $output_callback
+   * @param string $artifact_dir
+   * @param string $vcs_url
+   * @param string $vcs_path
+   *
+   * @throws \Acquia\Cli\Exception\AcquiaCliException
+   */
+  protected function prepareDir(Closure $output_callback, string $artifact_dir, string $vcs_url, string $vcs_path): void {
+    $fs = $this->localMachineHelper->getFilesystem();
+
+    $output_callback('out', "Removing $artifact_dir if it exists");
+    $fs->remove($artifact_dir);
+
+    $output_callback('out', "Initializing Git in $artifact_dir");
+    $process = $this->localMachineHelper->execute(['git', 'clone', '--depth', '1', '--branch', $vcs_path, $vcs_url, $artifact_dir], $output_callback, NULL, $this->output->isVerbose());
+    if (!$process->isSuccessful()) {
+      throw new AcquiaCliException('Failed to clone repository from the Cloud Platform: {message}', ['message' => $process->getErrorOutput()]);
+    }
+
+    $output_callback('out', 'Global .gitignore file is temporarily disabled during artifact builds.');
+    $this->localMachineHelper->execute(['git', 'config', '--local', 'core.excludesFile', 'false'], $output_callback, $artifact_dir, $this->output->isVerbose());
+    $this->localMachineHelper->execute(['git', 'config', '--local', 'core.fileMode', 'true'], $output_callback, $artifact_dir, $this->output->isVerbose());
+
+    // Vendor directories can be "corrupt" (i.e. missing scaffold files due to earlier sanitization) in ways that break composer install.
+    $output_callback('out', 'Removing vendor directories');
+    foreach ($this->vendorDirs($artifact_dir) as $vendor_directory) {
+      $fs->remove(Path::join($artifact_dir, $vendor_directory));
+    }
+  }
+
+  protected function build(Closure $output_callback, string $artifact_dir): void {
+    // @todo generate a deploy identifier
+    // @see https://git.drupalcode.org/project/drupal/-/blob/9.1.x/sites/default/default.settings.php#L295
+    $output_callback('out', "Mirroring source files from {$this->dir} to $artifact_dir");
```

Comment:
I think there’s a scenario in which this is problematic. If there is a new upstream change to the repository which has not yet been pulled locally, and then a user runs this command, it will effectively wipe out those changes. What is the desired behavior in that case?

it would be nice to check whether the local repository is up-to-date before initiating this build.